File: app/src/main/java/com/example/bloodbank/ui/activity/HomeActivity.java
package com.example.bloodbank.ui.activity;

import android.os.Bundle;
import android.view.MenuItem;

import androidx.annotation.NonNull;

import com.example.bloodbank.R;
import com.example.bloodbank.ui.fragment.homeCycle.EditProfileFragment;
import com.example.bloodbank.ui.fragment.homeCycle.HomeFragment;
import com.example.bloodbank.ui.fragment.homeCycle.MoreFragment;
import com.example.bloodbank.ui.fragment.homeCycle.NotificationFragment;
import com.google.android.material.bottomnavigation.BottomNavigationView;

import butterknife.BindView;
import butterknife.ButterKnife;

import static com.example.bloodbank.helper.HelperMethods.replace;

public class HomeActivity extends BaseActivity implements BottomNavigationView.OnNavigationItemSelectedListener {

    @BindView(R.id.activity_home_bn_bottom_navigation)
    BottomNavigationView activityHomeBnBottomNavigation;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);
        ButterKnife.bind(this);

        HomeFragment homeFragment = new HomeFragment();

        replace(homeFragment, getSupportFragmentManager(), R.id.activity_home_fl_content);
        activityHomeBnBottomNavigation.setOnNavigationItemSelectedListener(this);
    }

    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
        switch (item.getItemId()) {
            case R.id.main:
                HomeFragment homeFragment = new HomeFragment();
                replace(homeFragment, getSupportFragmentManager(), R.id.activity_home_fl_content);
                break;
            case R.id.profile:
                EditProfileFragment editProfileFragment = new EditProfileFragment();
                replace(editProfileFragment, getSupportFragmentManager(), R.id.activity_home_fl_content);
                break;
            case R.id.notification:
                NotificationFragment notificationFragment = new NotificationFragment();
                replace(notificationFragment, getSupportFragmentManager(), R.id.activity_home_fl_content);
                break;
            case R.id.more:
                MoreFragment moreFragment = new MoreFragment();
                replace(moreFragment, getSupportFragmentManager(), R.id.activity_home_fl_content);
                break;
        }
        return true;
    }

    @Override
    public void onBackPressed() {
        baseFragment.onBack();
    }
}

File: app/src/main/java/com/example/bloodbank/ui/activity/HomeActivity.kt
package com.example.bloodbank.ui.activity

import android.os.Bundle
import android.view.MenuItem
import androidx.databinding.DataBindingUtil
import butterknife.ButterKnife
import com.example.bloodbank.R
import com.example.bloodbank.databinding.ActivityHomeBinding
import com.example.bloodbank.extensions.replaceFragment
import com.example.bloodbank.ui.fragment.homeCycle.EditProfileFragment
import com.example.bloodbank.ui.fragment.homeCycle.HomeFragment
import com.example.bloodbank.ui.fragment.homeCycle.more.MoreFragment
import com.example.bloodbank.ui.fragment.homeCycle.notifications.NotificationFragment
import com.google.android.material.bottomnavigation.BottomNavigationView

class HomeActivity : BaseActivity(), BottomNavigationView.OnNavigationItemSelectedListener {

    private lateinit var binding: ActivityHomeBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = DataBindingUtil.setContentView(this, R.layout.activity_home)
        ButterKnife.bind(this)
        this.replaceFragment(R.id.activity_home_fl_content, HomeFragment())

        binding.activityHomeBnBottomNavigation.setOnNavigationItemSelectedListener(this)
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.main -> this.replaceFragment(R.id.activity_home_fl_content, HomeFragment())
            R.id.profile -> this.replaceFragment(R.id.activity_home_fl_content, EditProfileFragment())
            R.id.notification -> this.replaceFragment(R.id.activity_home_fl_content, NotificationFragment())
            R.id.more -> this.replaceFragment(R.id.activity_home_fl_content, MoreFragment())
        }
        return true
    }

    override fun onBackPressed() {
        baseFragment.onBack()
    }
}